### PR TITLE
Update PMD4 validations to reflect current muttnik expectations (#1249)

### PR DESCRIPTION
**Pull request by kieranlives**
- [x] Change DatasetExactlyOneContents -> DatasetMaxOneContents
- [x] Change DatasetExactlyOneGraph -> DatasetMinOneGraph
- [x] Remove DatasetExactlyOneLabel
- [x] Add DatasetExactlyOneTitle
- [x] Remove DatasetMaxOneContributor
- [x] Remove RecordExactlyOneLabel
- [x] Individual tests for modified/new validations
- [x] Run PMD4 suite on removed validations

Run tests with `clojure -M:test`

### CODE REVIEW

**Review by RickMoynihan — CHANGES_REQUESTED** [review]
Thanks for this @kieranlives.  

The changes look great, I've suggested some minor bike shedding fixes... but I'm very happy for this to be merged once they've been made.

File: test/pmd4/swirrl/validations/pmd4/SELECT_DatasetExactlyOneTitle.ttl [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+@prefix dct:    <http://purl.org/dc/terms/> .
+
+<http://test/title/0#fail> a pmdcat:Dataset .
+
+<http://test/title/1/single-title> a pmdcat:Dataset ;
+    dct:title "I have but one title!" .
+
+<http://test/title/1/duplicate-title> a pmdcat:Dataset ;
+    dct:title "I have the same title twice!" ;
+    dct:title "I have the same title twice!" .
```
[the remark this answers is not in the document]
Comment (kieranlives) [reply test/pmd4/swirrl/validations/pmd4/SELECT_DatasetExactlyOneTitle.ttl:11]: I wondered that! I was just following the query at this point that says `FILTER(?title1 != ?title2)` - is that similarly redundant + should I remove it as well?

[the remark this answers is not in the document]
Comment (RickMoynihan) [reply test/pmd4/swirrl/validations/pmd4/SELECT_DatasetExactlyOneTitle.ttl]: No it's not redundant, at least not in the same way...

this line in the sparql:

```sparql
            ?dataset dct:title ?title1, ?title2 .
```

Is just syntactic sugar for this:

```sparql
            ?dataset dct:title ?title1 .
            ?dataset dct:title ?title2 .
```

Which essentially just means bind `?title1` to the set of dataset titles, and also bind `?title2` to the set of dataset titles.  So at that point we have a table with a cartesian product in it:

  | dataset | title1 | title2 |
  | --- | --- | --- |
  | :some-ds | my title | my title | 
  | :some-ds | other title | other title |
  | :some-ds | my title | other title |
  | :some-ds | other title | my title |

So the line: 

`FILTER (?title1 != ?title2)`

then lets us convert that into:

  | dataset | title1 | title2 |
  | --- | --- | --- |
  | :some-ds | my title | other title | 

i.e. the `FILTER` is really there to stop us failing when we have just one title.

I guess we could probably remove it if we changed how the count worked.

**Review by RickMoynihan — CHANGES_REQUESTED** [review]

**Comment by RickMoynihan** [comment]
@Robsteranium I've been through each of these and they seem good.  And I've updated @kieranlives corresponding muttnik PR so it now passes wrt IC-19.

https://github.com/Swirrl/muttnik/pull/1327

This means once that PR is merged, muttnik fixtures will be aligned with these changes.

Are there any other comments on these changes?

**Comment by Robsteranium** [comment]
I've merged this in to #24.

Thanks!

**Pull request closed without merging**